uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/providers/IdentityProviderEndpointDocs.java

```diff
@@ -967,6 +967,47 @@ void getAllIdentityProviders() throws Exception {
                         responseFields));
     }
 
+    @Test
+    void getFilteredIdentityProviders() throws Exception {
+        Snippet responseFields = responseFields(
+            fieldWithPath("[].type").description("Type of the identity provider."),
+            fieldWithPath("[].originKey").description("Unique identifier for the identity provider."),
+            fieldWithPath("[].name").description(NAME_DESC),
+            fieldWithPath("[].config").description(CONFIG_DESCRIPTION),
+            fieldWithPath("[]." + FIELD_ALIAS_ID).description(ALIAS_ID_DESC).attributes(key("constraints").value("Optional")).optional().type(STRING),
+            fieldWithPath("[]." + FIELD_ALIAS_ZID).description(ALIAS_ZID_DESC).attributes(key("constraints").value("Optional")).optional().type(STRING),
+
+            fieldWithPath("[].version").description(VERSION_DESC),
+            fieldWithPath("[].active").description(ACTIVE_DESC),
+
+            fieldWithPath("[].id").description(ID_DESC),
+            fieldWithPath("[].identityZoneId").description(IDENTITY_ZONE_ID_DESC),
+            fieldWithPath("[].created").description(CREATED_DESC),
+            fieldWithPath("[].last_modified").description(LAST_MODIFIED_DESC)
+        );
+
+        mockMvc.perform(get("/identity-providers")
+                .param("rawConfig", "false")
+                .param("active_only", "false")
+                .param("origin", "my-oauth2-provider")
+                .header("Authorization", "Bearer " + adminToken)
+                .contentType(APPLICATION_JSON))
+            .andExpect(status().isOk())
+            .andDo(document("{ClassName}/{methodName}",
+                preprocessResponse(prettyPrint()),
+                requestHeaders(
+                    headerWithName("Authorization").description("Bearer token containing `zones.<zone id>.admin` or `zones.<zone id>.idps.read` or `uaa.admin` or `idps.read` (only in the same zone that you are a user of)"),
+                    headerWithName("X-Identity-Zone-Id").description("May include this header to administer another zone if using `zones.<zoneId>.admin` or `zones.<zone id>.idps.read` or `uaa.admin` scope against the default UAA zone.").optional(),
+                    IDENTITY_ZONE_SUBDOMAIN_HEADER
+                ),
+                requestParameters(
+                    parameterWithName("rawConfig").optional("false").type(BOOLEAN).description("Flag indicating whether the response should use raw, unescaped JSON for the `config` field of the IDP, rather than the default behavior of encoding the JSON as a string."),
+                    parameterWithName("active_only").optional("false").type(BOOLEAN).description("Flag indicating whether only active IdPs should be returned or all."),
+                    parameterWithName("origin").optional(null).type(STRING).description("<small><mark>UAA 77.10.0</mark></small> Return only IdPs with specific origin.")
```

Comment:
Is this `origin` the same thing as the `originKey` when you [create the IDPs](https://docs.cloudfoundry.org/api/uaa/version/77.8.0/#saml)? If so, why not just call this field `originKey`?

Reply:
ok done